packages/compat/src/inspector-support.mts

```diff
@@ -0,0 +1,120 @@
+// This script exposes the modules used by Ember Inspector so an app that
+// builds with Vite can be inspected.
+
+// Disable TypeScript because this is a static script we want to import as is.
+// @ts-nocheck
+
+import { macroCondition, dependencySatisfies } from '@embroider/macros';
+
+globalThis.emberInspectorLoader = {
```

Comment:
this is probably not the final solution, but it any case:
there where some discussions in the past around reducing globals. see https://github.com/emberjs/rfcs/issues/781
which is basically to support multiple ember apps in one page. Not sure how relevant that still is.

Also, there can be race conditions between page scripts loading and ember_debug scripts loading.
after this is setup, a `window.dispatchEvent(new Event("ember"))` is required. This is what is currently being done.

Reply:
Honnestly I don't think we will get read of the global in this context. It's just a compatibility piece that we are working on at the moment, something that makes bridge between the old world and Vite.

> there can be race conditions between page scripts loading and ember_debug scripts loading.
after this is setup, a window.dispatchEvent(new Event("ember")) is required. This is what is currently being used on ember_debug side.

I don't understand what I should conclude from that. Who is calling `window.dispatchEvent(new Event("ember"))` in the classic world?